src/lib/php/Dao/LicenseDao.php

```diff
@@ -713,4 +713,34 @@ public function getLicenseObligations($licenseLists, $candidate = false)
       return $ObligationRef;
     }
   }
+
+  /**
+   * @param string $licenseShortname
+   * @param int|null $groupId
+   * @return null
+   */
+  public function deleteLicenseByShortName($licenseShortname, $groupId=null)
+  {
+    $extraCondition = "";
+    $condition = 'rf_shortname=$1';
+    $param = array($licenseShortname);
+    $this->dbManager->deleteSingleRow(
+      "DELETE FROM license_ref WHERE $condition",
+      $param, __METHOD__ . ".$condition.only");
+    if (isset($groupId)) {
+      $userId = (isset($_SESSION) && array_key_exists('UserId', $_SESSION)) ? $_SESSION['UserId'] : 0;
+      if (!empty($userId)) {
+        $param[] = $userId;
+        $extraCondition = "AND group_fk IN (SELECT group_fk FROM group_user_member WHERE user_fk=$".count($param).")";
+      }
+      if (is_int($groupId) && empty($userId)) {
+        $param[] = $groupId;
+        $extraCondition = "AND group_fk=$".count($param);
+      }
+      $this->dbManager->deleteSingleRow(
+        "DELETE FROM license_candidate WHERE $condition $extraCondition",
+        $param, __METHOD__ . ".$condition.group");
+    }
+    return null;
```

Comment:
Instead of returning null, returning `true` if license was deleted and `false` otherwise will make more sense. Please have a look at `RETURNING` statement for Postgres.